python/hail/docs/_ext/hail_doctest.py

```diff
@@ -193,6 +194,29 @@ def summarize(self, out, verbose=None):
         out(string_io.getvalue())
         return res
 
+    def get_code(self):
+        return self.code
+
+    def run(self, test, compileflags=None, out=None, clear_globs=True):
+        original_optionflags = self.optionflags
+        for examplenum, example in enumerate(test.examples):
+            # Merge in the example's options.
+            self.optionflags = original_optionflags
```

Comment:
You need to either reset this after each iteration (at the bottom of the loop), or after the loop, too.  It looks like now you can leave with the updates modified by the last example.